src/Compilers/Core/Portable/SourceGeneration/GeneratorContexts.cs

```diff
@@ -139,6 +139,26 @@ public void RegisterForSyntaxNotifications(SyntaxReceiverCreator receiverCreator
             InfoBuilder.SyntaxReceiverCreator = receiverCreator;
         }
 
+        /// <summary>
+        /// Register a callback that is invoked after initialization.
+        /// </summary>
+        /// <remarks>
+        /// This method allows a generator to opt-in to an extra phase in the generator lifecycle called PostInitialization. After being initialized
+        /// any generators that have opted in will have their provided callback invoked with a <see cref="GeneratorPostInitializationContext"/> instance
+        /// that can be used to alter to the compilation that is provided to subsequent generator phases.
+        /// 
+        /// For example a generator may choose to add sources during PostInitialization. These will be added to the compilation before execution and
+        /// will be visited by a registered <see cref="ISyntaxReceiver"/> and available for semantic analysis as part of the <see cref="GeneratorExecutionContext.Compilation"/>
```

Comment:
Should probably add some clarification as to what is available at this stage in the lifecycle, why you should use it, and what you should use it for.